### PR TITLE
create BIMDataInputOutlined component and add documentation

### PR DESCRIPTION
**Pull request by LrxGaelle**
- [x] I have tested my component
- [x] I have updated the documentation or there is no documentation needed

## Libraries that use the DS (need to merge first)

- [ ] [BCF Components](https://github.com/bimdata/bcf-components)
- [ ] [BIMData Components](https://github.com/bimdata/bimdata-components)
- [ ] [Building maker](https://github.com/bimdata/building-maker)

## Repos that use the DS (and that I need to check after libraries have been merged)

- [ ] [Viewer](https://github.com/bimdata/viewer)
- [ ] [Platform](https://github.com/bimdata/platform)
- [ ] [SDK](https://github.com/bimdata/bimdata-viewer-sdk)
- [ ] [Marketplace](https://github.com/bimdata/marketplace-front)
- [ ] [Documentation](https://github.com/bimdata/documentation)

### CODE REVIEW

File: src/BIMDataComponents/BIMDataInputOutlined/BIMDataInputOutlined.vue [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      successAndError => {
+        if (successAndError) {
+          throw new Error("Input state cannot be both success and error.");
+        }
+      },
+    );
+  },
+  methods: {
+    focus() {
+      this.$refs.input && this.$refs.input.focus();
```
Comment (Kurtil) [inline src/BIMDataComponents/BIMDataInputOutlined/BIMDataInputOutlined.vue:173]: optional chaining ? :)

File: src/BIMDataComponents/BIMDataInputOutlined/BIMDataInputOutlined.vue [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    },
+    blur() {
+      this.$refs.input && this.$refs.input.blur();
```
Comment (Kurtil) [inline src/BIMDataComponents/BIMDataInputOutlined/BIMDataInputOutlined.vue:176]: optional chaining ? :)

File: src/web/views/Components/Input/props-data.js [hunk 3]
```diff
@@ -32,33 +14,51 @@ export default [
     "Use this boolean to disabled your input.",
   ],
   [
-    "loading",
+    "error",
     "Boolean",
     "false",
-    "",
+    "Use this boolean to check if your input is bad.",
```
Comment (Kurtil) [inline src/web/views/Components/Input/props-data.js:20]: `bad` ? ^^ `incorrect` may be more appropriate isn't it ?

File: src/web/views/Components/Input/props-data.js [hunk 4]
```diff
@@ -32,33 +14,51 @@ export default [
...
-    "Use this props to add a success message to your input.",
+    "Use this props to add a placeholder to your input.",
   ],
   [
-    "error",
+    "success",
     "Boolean",
     "false",
-    "Use this boolean to check if your input is bad.",
+    "Use this boolean to check if your input is good.",
```
Comment (Kurtil) [inline src/web/views/Components/Input/props-data.js:56]: `correct` ?

File: src/web/views/Components/Input/props-outlined-input.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    "disabled",
+    "Boolean",
+    "false",
+    "Use this boolean to disabled your input.",
+  ],
+  [
+    "error",
+    "Boolean",
+    "false",
+    "Use this boolean to check if your input is bad.",
```
Comment (Kurtil) [inline src/web/views/Components/Input/props-outlined-input.js:26]: idem

File: src/web/views/Components/Input/props-outlined-input.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    "placeholder",
+    "String",
+    "' '",
+    "Use this props to add a placeholder to your input.",
+  ],
+  [
+    "success",
+    "Boolean",
+    "false",
+    "Use this boolean to check if your input is good.",
```
Comment (Kurtil) [inline src/web/views/Components/Input/props-outlined-input.js:80]: idem

**Review by Kurtil — CHANGES_REQUESTED** [review]

**Review by NicolasRichel — APPROVED** [review]

**Pull request merged**